### DISCUSSION TITLE
IndexError: List index out of range

### DISCUSSION THREAD

**Discussion by Minglarn** [blakeblackshear/frigate#830] [General]
Dont think this is a bug but...
Trying to use from Corals web page a new edgetpu.tflite file (just for fun). The labelmap.txt is 1000+ objects big. 
From Corals web: https://coral.ai/models/
EfficientNet-EdgeTpu (L)
Recognizes 1,000 types of objects
Dataset: ImageNet
Input size: 300x300

The docker starts but exits after a minute. 
Starting the docker like this:
`docker run --rm -d --name frigate --privileged --shm-size=512m --mount type=tmpfs,target=/tmp/cache,tmpfs-size=1024M -v /dev/bus/usb:/dev/bus/usb -v /opt/frigate/clips:/media/frigate/clips -v /opt/frigate/recordings:/media/frigate/recordings -v /opt/frigate/config:/config:ro -v /opt/frigate/labelmap.txt:/labelmap.txt -v /opt/frigate/edgetpu_model.tflite:/edgetpu_model.tflite -v /etc/localtime:/etc/localtime:ro -e FRIGATE_RTSP_PASSWORD='password' -p 5000:5000 -p 1935:1935 blakeblackshear/frigate:0.8.3-amd64
`
Error dump:
`
frigate.mqtt                   INFO    : MQTT connected

detector.coral                 INFO    : Starting detection process: 36

frigate.app                    INFO    : Camera processor started for frigate_entren: 40

frigate.edgetpu                INFO    : Attempting to load TPU as usb:0

frigate.app                    INFO    : Camera processor started for frigate_parkering: 41

frigate.edgetpu                INFO    : TPU found

frigate.app                    INFO    : Camera processor started for frigate_mot_parkering: 43

frigate.app                    INFO    : Camera processor started for frigate_tomten: 44

frigate.app                    INFO    : Camera processor started for frigate_tapo: 46

frigate.app                    INFO    : Camera processor started for frigate_pathway: 48

frigate.app                    INFO    : Camera processor started for frigate_pryttelverkstad: 50

frigate.app                    INFO    : Camera processor started for frigate_lillstugan: 51

frigate.app                    INFO    : Camera processor started for frigate_baksida: 52

frigate.app                    INFO    : Capture process started for frigate_entren: 53

frigate.app                    INFO    : Capture process started for frigate_parkering: 56

frigate.app                    INFO    : Capture process started for frigate_mot_parkering: 59

frigate.app                    INFO    : Capture process started for frigate_tomten: 64

frigate.app                    INFO    : Capture process started for frigate_tapo: 67

frigate.app                    INFO    : Capture process started for frigate_pathway: 70

frigate.app                    INFO    : Capture process started for frigate_pryttelverkstad: 73

frigate.app                    INFO    : Capture process started for frigate_lillstugan: 77

frigate.app                    INFO    : Capture process started for frigate_baksida: 79

Process detector:coral:

Traceback (most recent call last):

  File "/usr/lib/python3.8/multiprocessing/process.py", line 315, in _bootstrap

    self.run()

  File "/usr/lib/python3.8/multiprocessing/process.py", line 108, in run

    self._target(*self._args, **self._kwargs)

  File "/opt/frigate/frigate/edgetpu.py", line 150, in run_detector

    detections = object_detector.detect_raw(input_frame)

  File "/opt/frigate/frigate/edgetpu.py", line 100, in detect_raw

    label_codes = np.squeeze(self.interpreter.get_tensor(self.tensor_output_details[1]['index']))

IndexError: list index out of range

frigate.watchdog               INFO    : Detection appears to have stopped. Exiting frigate...

frigate.app                    INFO    : Stopping...

frigate.object_processing      INFO    : Exiting object processor...

frigate.events                 INFO    : Exiting event processor...

frigate.events                 INFO    : Exiting event cleanup...

frigate.watchdog               INFO    : Exiting watchdog...

frigate.record                 INFO    : Exiting recording maintenance...

frigate.stats                  INFO    : Exiting watchdog...

peewee.sqliteq                 INFO    : writer received shutdown request, exiting.

root                           INFO    : Waiting for detection process to exit gracefully...
`

Both labelmap and edgetpu file is located where frigate docker can find them. 
Any help would be appreciated.

**Comment by blakeblackshear** [comment]
Looks like you are using an Image classification model rather than an object detection model. That won't work. The inputs and outputs much be exactly what frigate expects.

**Reply by Minglarn** [reply]
Yes... Saw that later that it was a "Image" class instead of "Object" class... 
Thanks...
